src/js/components/stock-movement-wizard/inbound/SendMovementPage.jsx

```diff
@@ -764,6 +764,20 @@ class SendMovementPage extends Component {
                         if (document.hidden) {
                           return null;
                         }
+
+                        // Some documents have multiple download options so list them all out.
+                        if (document.downloadOptions) {
+                          return _.map(document.downloadOptions, (downloadOption) => (
+                            <DocumentButton
+                              link={downloadOption.uri}
+                              buttonTitle={downloadOption.name}
+                              {...document}
+                              key={idx}
```

Comment:
index shouldn't be a key. Moreover - this map is nested, hence you will have the same key for those elements.

Reply:
why is it bad to make it the index? I don't think this array is mutable (and so indexes don't change). It just uses data returned from the backend.

I suppose I could change the key to be the document/download name

Reply:
https://www.youtube.com/watch?v=xlPxnc5uUPQ

+ the key should be unique. since you have nested `.map`, the `idx` will be the same for each download option (you use` idx` from the nested map)

Reply:
yes I understand that part and I'll fix that to make the index actually unique.

I watched the video and it makes sense (thanks for sharing). I was wondering if it's fine to use index in this case since the array shouldn't be changing, but actually I'm not certain about it anymore since there's the ability to add/remove documents from this page. I don't know what the behaviour is in that situation.

But that's a pre-existing concern since this was already using index as the id. And there isn't a better option right now in the payload IMO. The only other possible field to use is name (which is translated and not guaranteed to be unique). I think I'm going to make it work with ids as it was before and add a TODO to fix it to provide a proper unique id.